chain/chain/src/resharding/event_type.rs

```diff
@@ -26,34 +26,35 @@ pub struct ReshardingSplitShardParams {
     pub right_child_shard: ShardUId,
     /// The account at the boundary between the two children.
     pub boundary_account: AccountId,
-    /// Hash of the first block having the new shard layout.
+    /// Hash of the last block having the old shard layout.
     pub block_hash: CryptoHash,
     /// The block before `block_hash`.
     pub prev_block_hash: CryptoHash,
 }
 
 impl ReshardingEventType {
-    /// Takes as input a [ShardLayout] definition and deduces which kind of resharding operation must be
-    /// performed.
+    /// Takes as input a [ShardLayout] definition and deduces which kind of resharding operation
+    /// must be performed.
     ///
     /// # Args:
-    /// * `shard_layout`: the new shard layout
-    /// * `block_hash`: hash of the first block with `shard_layout`
+    /// * `next_shard_layout`: the new shard layout
+    /// * `block_hash`: hash of the last block with the shard layout before `next_shard_layout`
```

Comment:
mini nit: `with the old shard layout` is less verbose and hopefully still clear what it means. Feel free to leave as is if you prefer the more precise one.